source/module_cell/parallel_kpoints.h

```diff
@@ -37,24 +37,38 @@ class Parallel_Kpoints
 #endif
 
     // information about pool, dim: GlobalV::KPAR
-    int* nproc_pool = nullptr;
-    int* startpro_pool = nullptr;
+    //int* nproc_pool = nullptr;    it is not used
+    //int* startpro_pool = nullptr;
 
     // inforamation about kpoints, dim: GlobalV::KPAR
-    int* nks_pool = nullptr;    // number of k-points in each pool
-    int* startk_pool = nullptr; // the first k-point in each pool
-    int kpar = 0;               // number of pools
+    std::vector<int> nks_pool;    // number of k-points in each pool
+    std::vector<int> startk_pool; // the first k-point in each pool
 
     // information about which pool each k-point belongs to,
-    int* whichpool = nullptr; // whichpool[k] : the pool which k belongs to, dim: nkstot_np
+    std::vector<int> whichpool; // whichpool[k] : the pool which k belongs to, dim: nkstot_np
+    
     int nkstot_np = 0;        // number of k-points without spin, kv.set_nkstot(nkstot_np) * nspin(1 or 2)
     int nks_np = 0;           // number of k-points without spin in the present pool
+    
+    int get_startpro_pool(const int& pool) const
```

Comment:
Public interface in class should have annotation , which should be doxygen format(/// or /**...*/)

Reply:
New code should follow this criteria, existed code can be skipped